client/me/notification-settings/blogs-settings/index.jsx

```diff
@@ -1,74 +1,97 @@
 /**
  * External dependencies
  */
-import React, { PropTypes } from 'react';
+import React, { Component, PropTypes } from 'react';
+import { connect } from 'react-redux';
+import { localize } from 'i18n-calypso';
+import { noop } from 'lodash';
 import Immutable from 'immutable';
 
 /**
  * Internal dependencies
  */
+import { getSites } from 'state/selectors';
 import EmptyContentComponent from 'components/empty-content';
 import Blog from './blog';
 import InfiniteList from 'components/infinite-list';
 import Placeholder from './placeholder';
+import QuerySites from 'components/data/query-sites';
 import config from 'config';
 
-export default React.createClass( {
-	displayName: 'BlogsSettings',
+const createPlaceholder = () => <Placeholder />;
 
-	propTypes: {
-		blogs: PropTypes.object.isRequired,
+const getItemRef = ( { ID } ) => `blog-${ ID }`;
+
+class BlogsSettings extends Component {
+	static propTypes = {
+		sites: PropTypes.array.isRequired,
 		devices: PropTypes.object.isRequired,
 		settings: PropTypes.instanceOf( Immutable.List ),
 		hasUnsavedChanges: PropTypes.bool.isRequired,
 		onToggle: PropTypes.func.isRequired,
 		onSave: PropTypes.func.isRequired,
 		onSaveToAll: PropTypes.func.isRequired
-	},
+	};
 
 	render() {
-		if ( ! this.props.blogs.initialized || ! this.props.devices.initialized || ! this.props.settings ) {
-			return <Placeholder />;
+		const { sites, translate } = this.props;
+
+		if ( ! sites || ! this.props.devices.initialized || ! this.props.settings ) {
+			return (
+				<div>
+					<QuerySites allSites />
```

Comment:
Absolutely right. This is not necessary. I must have had a weird state on my browser. Fixed on bb80c9cfd49913748d417d2974508abc8cd26668